include/net/buf.h

```diff
@@ -1431,19 +1433,19 @@ static inline void *net_buf_user_data(const struct net_buf *buf)
 }
 
 /**
- * @def net_buf_reserve
  * @brief Initialize buffer with the given headroom.
  *
  * The buffer is not expected to contain any data when this API is called.
  *
  * @param buf Buffer to initialize.
  * @param reserve How much headroom to reserve.
  */
-#define net_buf_reserve(buf, reserve) net_buf_simple_reserve(&(buf)->b, \
-							     reserve)
+static inline void net_buf_reserve(struct net_buf *buf, size_t reserve)
+{
+	net_buf_simple_reserve(&(buf)->b, reserve);
```

Comment:
One more review iteration, hopefully your editor-automation skills are good enough that you're not doing these updates manually to every single place :) 

The parenthesis around `(buf)` was really only needed because of macro expansion, i.e. to ensure correct order of evaluation if something else than an "atomic" input parameter was given (say, `net_buf_reserve(array + index, reserve)`). Now that `buf` is a real variable this is no longer necessary.

Reply:
They could be better, but identical data replacement is easy 👍